mono/profiler/proflog.c

```diff
@@ -3054,6 +3056,11 @@ get_file_content (FILE *stream)
 
 	while ((bytes_read = fread (buffer, 1, LINE_BUFFER_SIZE, stream)) > 0) {
 		g_string_append_len (builder, buffer, bytes_read);
+
+		if (builder->len > MAX_FILE_SIZE) {
```

Comment:
I think it'd be best to get the file size with `fseek / ftell`. Then you could also pre-size the string with the right size.